### PR TITLE
Preciso Bid Adapter : update on valid request checks 

### PR DESCRIPTION
**Pull request by NikhilGopalChennissery**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by patmmccann** [comment]
It appears your duplication fixes were tricks to avoid the report. Please actually fix

File: modules/precisoBidAdapter.js [hunk 1]
```diff
@@ -1,147 +1,150 @@
...
+    //  req.language.indexOf('-') != -1 && (req.language = req.language.split('-')[0])
     if (bidderRequest) {
       if (bidderRequest.uspConsent) {
-        request.ccpa = bidderRequest.uspConsent;
+        req.ccpa = bidderRequest.uspConsent;
       }
       if (bidderRequest.gdprConsent) {
-        request.gdpr = bidderRequest.gdprConsent
+        req.gdpr = bidderRequest.gdprConsent
       }
```
Comment (patmmccann) [inline modules/precisoBidAdapter.js:101]: There are many tricks to fool the duplication detector in this pr. Remove them.

**Review by patmmccann — CHANGES_REQUESTED** [review]

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann 
Sorry for the changes we have made. We are beginners on the Prebid platform. We have tested the current code changes locally and confirmed that it is working fine.

We created our Adapter by referring to the documentation on prebid.org and examining other bidAdapters' code as well. We have  noticed that some of  codes is common in some other adapters as well. 

I am ready to modify the code as you suggested. I would like to clarify a few points:

We are accessing storage manager from the BidAdapter itself. Is that okay?
For building Bid requests and Bid responses, should I create a common library function and use that library in our bid adapter.?

Please correct me if I am wrong.

 Thank you.

**Comment by patmmccann** [comment]
You may access storage manager, however you will not have access by default, publishers must grant access. 

Yes, importing functions from common libraries should reduce code duplication. Your particular adapter looks similar to at least one adapter maintained by teqblaze, who I believe has that effort underway. You may wish to coordinate with them or look at their patterns

https://github.com/MaksymTeqBlaze/Prebid.js/commit/0249eb7a258d610e937a114676e24c1d3f6c956e

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann ,

Thanks for your update. I hope this will help us  to remove the code duplications

We will work on this and update the PR ASAP.

Thank you!

**Comment by NikhilGopalChennissery** [comment]
Hi @patmmccann ,

I have created a new PR ( https://github.com/prebid/Prebid.js/pull/11868 )  with our latest changes to remove the code duplication.

**Pull request closed without merging**